### PR TITLE
#17 - Create Navlink component

### LINKED ISSUE #17: Create Navlink component
- Reuse Box, IconSvg, Text

Props: 
- onClick
- isActive

image

### PR DESCRIPTION
**Pull request by chanchal-sadhukhan**
## Description
Navlink Component added

## Screenshot
Screenshot from 2021-09-02 20-36-23

## Platform tested
- [ ] Large
- [ ] Tablet
- [ ] Mobile

### CODE REVIEW

File: src/components/primitive/navlink/navlink.component.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { withTheme } from 'styled-components';
+import { IStyledTheme } from '../../../themes/styled-components';
+import { NavlinkComponentProps } from './navlink.component.props';
+import { Box } from '../box/box.component';
+import { Text } from '../text/text.component';
+import { IconSvg } from '../icon-svg/icon-svg.component';
+
+// @ts-ignore
```
Comment (dsouzaedison) [inline src/components/primitive/navlink/navlink.component.tsx]: Specify reason for ts-ignore always

File: src/components/primitive/navlink/navlink.component.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    const {
+      theme: { colors },
+      active,
+      label,
+      icon,
+      ...rest
+    } = props;
+
+    return (
+      <Box
```
Comment (dsouzaedison) [inline src/components/primitive/navlink/navlink.component.tsx]: What happens when you click this link?

**Review by dsouzaedison — CHANGES_REQUESTED** [review]

File: src/components/primitive/navlink/navlink.component.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  return (
+    // @ts-ignore - No overload matches this call
+    <NavlinkContainer
+      row
+      hCenter
+      margin='tiny'
+      paddingHorizontal='small'
+      paddingVertical='tiny'
+      color={active ? 'textLighter' : 'white'}
+      onClick={onClick}
```
Comment (dsouzaedison) [inline src/components/primitive/navlink/navlink.component.tsx:28]: Does box support onClick?

Comment (chanchal-sadhukhan) [reply src/components/primitive/navlink/navlink.component.tsx:28]: Yes

**Review by dsouzaedison — APPROVED** [review]

**Pull request merged**